### PR TITLE
Plugin/Preset Ordering

### PR DESCRIPTION
**Pull request by hzoo**

### CODE REVIEW

**Comment by danez** [comment]
Are plugins applied before or after presets?

**Comment by hzoo** [comment]
Yeah plugin before presets, and ordering is based on vistors used in the plugins (I guess this would make sense because we are doing the merging of visitors for the single pass) @taion

**Comment by taion** [comment]
Another thing maybe worth noting is that if you do something like https://github.com/facebook/jest/blob/v13.2.3/packages/babel-jest/src/index.js#L17-L22, the presets (and plugins) specified there will all run before anything specified in `.babelrc`.

This came up in https://github.com/facebook/jest/issues/1387#issuecomment-238708769.

File: docs/plugins.md [hunk 1]
```diff
@@ -171,6 +171,34 @@ These plugins allow Babel to parse specific types of syntax.
...
+
+Preset Ordering is reversed (last to first).
+
+> Yes this is confusing, see [babel/notes #2](https://github.com/babel/notes/blob/master/2016-08/august-01.md#potential-api-changes-for-traversal).
+
+> I believe the reason why (for backwards compatability) is that most users had listed "es2015" first and "stage-0" second.
+> And stage-0 would run before es2015.
+
+```js
+"plugins": [
```
Comment (ykzts) [inline docs/plugins.md]: Did you mean "`presets`"?

**Pull request merged**